### PR TITLE
`KnownTypes` in Dart

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR enhances the Spine Dart client library with Dart code generation. Now, instead of accepting  `Any`s, the API accepts `GeneratedMessage`s. This allows for fewer errors on the API users' side.

Also, in this PR we improve the documentation of the library keeping in mind the next stage of development — publishing the Dart package to `pub`.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: build.gradle [hunk 1]
```diff
@@ -63,6 +63,7 @@ subprojects {
 
         dependencies {
             classpath deps.build.guava
+            classpath "io.spine.tools:spine-proto-dart-plugin:$spineBaseVersion"
```
Comment (armiol) [inline build.gradle]: Let's move it closer to its audience.

File: client-dart/lib/spine_client.dart [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/// The central library of this package.
```
Comment (armiol) [inline client-dart/lib/spine_client.dart]: It looks an "entry point", not a "central library".  The central library looks like [this](https://media.glassdoor.com/l/06/df/0d/b8/amsterdam-public-library.jpg).

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

**Pull request merged**